tests/nativeapp/codegen/test_sandbox.py

```diff
@@ -49,8 +46,15 @@ def mock_environ():
 
 
 @pytest.fixture
-def fake_venv_root(temp_dir):
-    venv_root = Path(temp_dir) / "venv"
+def fake_venv_root(fake_venv_root_unix, fake_venv_root_win32):
+    if IS_WINDOWS:
+        return fake_venv_root_win32
+    return fake_venv_root_unix
```

Comment:
Can you explain what the actual issue was? These tests were set up to mimic the file/dir structure, but really shouldn't have relied on the actual OS behaviour. As updated, we seem to lose the ability to run across operating systems, which was part of the original intention.

Reply:
the path that the CLI looks for is different based on the value of `sys.platform`: https://github.com/snowflakedb/snowflake-cli/blob/a0ad54f90b62173857a8c6d7df8c2f7f6b089d26/src/snowflake/cli/plugins/nativeapp/codegen/sandbox.py#L106-L109

Reply:
the reason these tests were failing was because we were creating a temporary fake venv with `<root>/bin/python` when the CLI was looking for `<root>/Scripts/python.exe` in some of the tests that don't patch `sys.platform`

Reply:
Ok, that makes sense now, thanks.